sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java

```diff
@@ -530,6 +544,16 @@ interface WithHibernationSupport {
              */
             Update withoutHibernationSupport();
         }
+
+        /** The stage of the managed disk update allowing to specify hypervisor generation. */
+        interface WithHyperVGeneration {
+            /**
+             * Specifies the hypervisor generation.
+             * @param hyperVGeneration the hypervisor generation
+             * @return the next stage of the update
+             */
+            Update withHyperVGeneration(HyperVGeneration hyperVGeneration);
```

Comment:
do we need without API? `withoutHyperVGeneration()`

Reply:
This fix is to enable managed disks to support `HyperVGeneration` in create and update using convenience method.

Reply:
it is related to what this feature is used for, can a user drop this setting (ie set it to null). We can add later if needed.